src/foundry/client-esm/canvas/tokens/ring.d.mts

```diff
@@ -1,5 +1,180 @@
-declare class TokenRing {}
+/**
+ * Dynamic Token Ring Manager.
+ */
+declare class TokenRing {
+  #TokenRing: true;
 
-declare namespace TokenRing {}
+  /** A `TokenRing` is constructed by providing a reference to a `Token` object. */
+  constructor(token: Token.ConfiguredInstance);
+
+  /** The effects which could be applied to a token ring (using bitwise operations) */
+  static effects: TokenRing.Effects;
+
+  /** Is the token rings framework enabled? Will be `null` if the system hasn't initialized yet. */
+  static get initialized(): boolean | null;
+
+  /** Token Rings sprite sheet base texture. */
+  static baseTexture: PIXI.BaseTexture;
+
+  /** Rings and background textures UVs and center offset. */
+  static texturesData: Record<string, { UVs: Float32Array; center: { x: number; y: number } }>;
+
+  /** The token ring shader class definition. */
+  static tokenRingSamplerShader: typeof TokenRingSamplerShader;
```

Comment:
Is this set in the CONFIG in any way? If so it should refer to the config directly.

Can this be overriden with a different constructor? If so it should have an `AnyConstructor` added.